app/controllers/charges_controller.rb

```diff
@@ -15,16 +15,29 @@ def create
 
     seller_id = charge_params[:seller_id]
 
-    is_distribution = charge_params[:is_distribution] || false
+    campaign = if charge_params[:is_distribution].present?
+      # TODO(justintmckibben): Delete this case when we start using campaign_id
+      #                        in the frontend
+      Campaign.find_by(
+        seller_id: seller.id,
+        active: true,
+        valid: true
+      )
+    elsif charge_params[:campaign_id].present?
+      Campaign.find_by(campaign_id: campaign_id)
+    end
 
-    validate(seller_id: seller_id, line_items: line_items, is_distribution: is_distribution)
+    validate(
+      seller_id: seller_id,
+      line_items: line_items,
+      gift_a_meal: campaign.present?
```

Comment:
love how we're changing `is_distribution` to `gift_a_meal`, way more understandable